SOURCES/etc/xapi.d/plugins/updater.py

```diff
@@ -106,44 +106,30 @@ def display_package(p):
     return {'name': p.name, 'version': p.version, 'release': p.release, 'description': p.summary,
             'changelog': changelog, 'url': p.url, 'size': p.size, 'license': p.license}
 
-def build_repo_list(additional_repos):
+def build_repo_list(enabled_repos, additional_repos):
     repos = list(DEFAULT_REPOS)
     if additional_repos:
         repos += [x.strip() for x in additional_repos.split(',')]
-    return repos
+    return filter(lambda repo: repo in [x.id for x in enabled_repos], repos)
+
+def install_helper(session, args, mode):
```

Comment:
It's called `install_helper` but also handles updates, which is slightly confusing. I wouldn't say updates are a kind of install, not in yum's vocabulary at least. However I don't have an obvious replacement (you wouldn't like `update_or_install` , right?), so :man_shrugging: 

Also, a bit of nitpicking, but I'd rename `mode` to `action`, which better describes what the parameter means.

Reply:
From my pov, an update is just an installation that replaces an existing package. So I don't see a major issue using this naming.

Regarding action, it's a good choice.